eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/Utils.java

```diff
@@ -0,0 +1,46 @@
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Common utils amount custom checks
+ */
+public class Utils {
+    private static final Set INVALID_FINAL_COMBINATION = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+        TokenTypes.LITERAL_TRANSIENT,
+        TokenTypes.LITERAL_VOLATILE
+    )));
+
+    private static final Set INVALID_FINAL_ANNOTATIONS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+        "JsonProperty"
+    )));
+
+    /**
+     * Check if variable modifiers contains any of the illegal combination with final modifier
+     * For instance, we don't want to combine transient or volatile with final
+     *
+     * @param modifiers a DetailAST pointing to a Variable list of modifiers
+     * @return true if there is any modifier that shouldn't be combined with final
+     */
+    protected static boolean hasIllegalCombination(DetailAST modifiers) {
+        for (DetailAST modifier = modifiers.getFirstChild(); modifier != null; modifier = modifier.getNextSibling()) {
```

Comment:
For this util method, the javadocs specifies this should be for a variable node, but you never do any checks to ensure that this isn't some random node like CLASS_DEF.

Also, why protected?

Reply:
can be replaced with: `TokenUtil.findFirstTokenByPredicate(...)`. I think this could be simplified.
```java
node -> {
    int modifierType = modifier.getType();
    return INVALID_FINAL_COMBINATION.contains(modifierType)
        || (TokenTypes.ANNOTATION == modifierType && INVALID_FINAL_ANNOTATIONS.contains(modifier.findFirstToken(TokenTypes.IDENT).getText()));
}
```

Reply:
@conniey , thanks for the comments!
- protected to have it only available for same package `com.azure.tools.checkstyle.checks`
No need to use it outside from there. All custom checks are added to this package.

- I like your proposal with `findFirstTokenByPredicate()` . I will update it.